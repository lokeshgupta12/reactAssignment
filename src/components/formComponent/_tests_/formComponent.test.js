import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import FormComponent from '../index';
import Adapter from 'enzyme-adapter-react-16';

// Configure Enzyme adapter for Jest
Enzyme.configure({ adapter: new Adapter() });

describe('<FormComponent /> component', () => {

  it('should match the snapshot', () => {
    const tree = renderer.create(<FormComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the HTML Correctly', () => {
    const wrapper = shallow(<FormComponent />);

    expect(wrapper.find('div').length).toBe(4);
  });

  it('should render the HTML Correctly', () => {
    const wrapper = shallow(<FormComponent />);

    expect(wrapper.find('InputControl').length).toBe(2);
  });

  it('should render the HTML Correctly', () => {
    const wrapper = shallow(<FormComponent />);

    expect(wrapper.find('button').length).toBe(2);
  });
})