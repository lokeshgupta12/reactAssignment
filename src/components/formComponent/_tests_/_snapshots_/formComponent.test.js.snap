// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormComponent /> component should match the snapshot 1`] = `
  <div className="col-md-3">
            <div className="form-group">
                <InputControl
                    onChange={[Function]}
                    />
            </div>
            <div className="form-group">
            <InputControl
                onChange={[Function]}
            />
            </div>
            <button className="btn btn-primary" onClick={[Function]} >Submit</button>
            <button className="btn btn-secondary resetBtn" onClick={[Function]}>Reset</button>
        </div>
        <div className="col-md-9">
            <MapWithADirectionsRenderer 
            ></MapWithADirectionsRenderer>
        </div>
        <LoadingSpinner  enableLoading = true/>
`;